sdk/dotnet/Pulumi/Deployment/Deployment_RegisterResource.cs

```diff
@@ -21,6 +21,10 @@ public partial class Deployment
             var label = $"resource:{name}[{type}]";
             Log.Debug($"Registering resource start: t={type}, name={name}, custom={custom}, remote={remote}");
 
+            if (options.DeletedWith != null && await MonitorSupportsDeletedWith().ConfigureAwait(false)) {
```

Comment:
Missing a bool negation here
```suggestion
            if (options.DeletedWith != null && !(await MonitorSupportsDeletedWith().ConfigureAwait(false))) {
```